### PR TITLE
ipq806x: add nor config wpq864

### LINKED ISSUE #15476: compex wpq864 Unable to mount root fs
### Describe the bug

Boot openwrt-ipq806x-generic-compex_wpq864-initramfs-fit-uImage.itb via tftpboot.
run sysupgrade -n openwrt-ipq806x-generic-compex_wpq864-squashfs-nand-sysupgrade.bin
next reboot halted with kernelpanic "Unable to mount root fs"

```
[    0.282485] nand: device found, Manufacturer ID: 0x2c, Chip ID: 0xaa
[    0.285837] qcom-pcie 1b900000.pci: host bridge /soc/pci@1b900000 ranges:
[    0.294021] nand: Micron MT29F2G08ABBEAH4
[    0.305370] nand: 256 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    0.311356] qcom-pcie 1b900000.pci:       IO 0x0035e00000..0x0035e0ffff -> 0x0000000000
[    0.317738] 23 fixed-partitions partitions found on MTD device qcom_nand.0
[    0.321922] qcom-pcie 1b900000.pci:      MEM 0x0032000000..0x0035dfffff -> 0x0032000000
[    0.325570] Creating 23 MTD partitions on "qcom_nand.0":
[    0.350355] 0x000000000000-0x000000040000 : "0:SBL1"
[    0.355004] qcom-pcie 1b700000.pci: iATU: unroll F, 8 ob, 8 ib, align 4K, limit 4G
[    0.363944] 0x000000040000-0x000000180000 : "0:MIBIB"
[    0.379876] 0x000000180000-0x0000002c0000 : "0:SBL2"
[    0.383172] qcom-pcie 1b500000.pci: iATU: unroll F, 8 ob, 8 ib, align 4K, limit 4G
[    0.400696] 0x0000002c0000-0x000000540000 : "0:SBL3"
[    0.462279] 0x000000540000-0x000000660000 : "0:DDRCONFIG"
[    0.467918] qcom-pcie 1b900000.pci: iATU: unroll F, 8 ob, 8 ib, align 4K, limit 4G
[    0.475056] 0x000000660000-0x000000780000 : "0:SSD"
[    1.037951] 0x000000780000-0x000000a00000 : "0:TZ"
[    1.044947] 0x000000a00000-0x000000c80000 : "0:RPM"
[    1.049955] 0x000000c80000-0x000001180000 : "0:APPSBL"
[    1.058911] 0x000001180000-0x000001200000 : "0:APPSBLENV"
[    1.060435] 0x000001200000-0x000001340000 : "0:ART"
[    1.066213] 0x000001340000-0x000005340000 : "ubi"
[    1.174931] 0x000005340000-0x0000053a0000 : "0:BOOTCONFIG"
[    1.176303] 0x0000053a0000-0x0000054e0000 : "0:SBL2_1"
[    1.182059] 0x0000054e0000-0x000005760000 : "0:SBL3_1"
[    1.189360] 0x000005760000-0x000005880000 : "0:DDRCONFIG_1"
[    1.192090] 0x000005880000-0x0000059a0000 : "0:SSD_1"
[    1.197646] 0x0000059a0000-0x000005c20000 : "0:TZ_1"
[    1.205235] 0x000005c20000-0x000005ea0000 : "0:RPM_1"
[    1.210132] 0x000005ea0000-0x000005f00000 : "0:BOOTCONFIG1"
[    1.211600] 0x000005f00000-0x000006400000 : "0:APPSBL_1"
[    1.224646] 0x000006400000-0x00000a400000 : "ubi_1"
[    1.331237] 0x00000a400000-0x000010000000 : "unused"
[    1.381408] qcom-pcie 1b700000.pci: Phy link never came up
[    1.382032] qcom-pcie 1b700000.pci: PCI host bridge to bus 0001:00
[    1.385929] pci_bus 0001:00: root bus resource [bus 00-ff]
[    1.391978] pci_bus 0001:00: root bus resource [io  0x10000-0x1ffff] (bus address [0x0000-0xffff])
[    1.397535] pci_bus 0001:00: root bus resource [mem 0x2e000000-0x31dfffff]
[    1.406583] pci 0001:00:00.0: [17cb:0101] type 01 class 0x060400
[    1.413417] pci 0001:00:00.0: supports D1
[    1.419495] pci 0001:00:00.0: PME# supported from D0 D1 D3hot
[    1.429942] PCI: bus0: Fast back to back transfers disabled
[    1.430460] PCI: bus1: Fast back to back transfers enabled
[    1.434528] pci 0001:00:00.0: PCI bridge to [bus 01-ff]
[    1.441085] qcom-pcie 1b500000.pci: Phy link never came up
[    1.445461] qcom-pcie 1b500000.pci: PCI host bridge to bus 0000:00
[    1.450660] pci_bus 0000:00: root bus resource [bus 00-ff]
[    1.456890] pci_bus 0000:00: root bus resource [io  0x0000-0xffff]
[    1.462285] pci_bus 0000:00: root bus resource [mem 0x08000000-0x0fdfffff]
[    1.468633] pci 0000:00:00.0: [17cb:0101] type 01 class 0x060400
[    1.474670] random: crng init done
[    1.475555] pci 0000:00:00.0: supports D1
[    1.484753] pci 0000:00:00.0: PME# supported from D0 D1 D3hot
[    1.495262] PCI: bus0: Fast back to back transfers disabled
[    1.495815] PCI: bus1: Fast back to back transfers enabled
[    1.499980] pci 0000:00:00.0: PCI bridge to [bus 01-ff]
[    1.506018] qcom-pcie 1b900000.pci: Phy link never came up
[    1.510872] qcom-pcie 1b900000.pci: PCI host bridge to bus 0002:00
[    1.516229] pci_bus 0002:00: root bus resource [bus 00-ff]
[    1.522271] pci_bus 0002:00: root bus resource [io  0x20000-0x2ffff] (bus address [0x0000-0xffff])
[    1.527885] pci_bus 0002:00: root bus resource [mem 0x32000000-0x35dfffff]
[    1.536869] pci 0002:00:00.0: [17cb:0101] type 01 class 0x060400
[    1.543701] pci 0002:00:00.0: supports D1
[    1.549765] pci 0002:00:00.0: PME# supported from D0 D1 D3hot
[    1.560691] PCI: bus0: Fast back to back transfers disabled
[    1.561352] PCI: bus1: Fast back to back transfers enabled
[    1.565280] pci 0002:00:00.0: PCI bridge to [bus 01-ff]
[    1.571283] spi_qup 1a280000.spi: IN:block:16, fifo:64, OUT:block:16, fifo:64
[    1.571750] pcieport 0001:00:00.0: AER: enabled with IRQ 34
[    1.575990] flash@0 enforce active low on GPIO handle
[    1.590251] pcieport 0000:00:00.0: AER: enabled with IRQ 37
[    1.594774] spi-nor spi0.0: s25fl256s1 (32768 Kbytes)
[    1.594962] 2 fixed-partitions partitions found on MTD device spi0.0
[    1.604233] Creating 2 MTD partitions on "spi0.0":
[    1.610753] 0x000000000000-0x000001000000 : "rootfs"
[    1.616305] pcieport 0002:00:00.0: AER: enabled with IRQ 39
[    1.616371] mtd: setting mtd23 (rootfs) as root device
[    1.625999] mtdsplit: no squashfs found in "rootfs"
[    1.630905] 0x000001000000-0x000002000000 : "scratch"
[    1.645830] ipq806x-gmac-dwmac 37000000.ethernet: IRQ eth_wake_irq not found
[    1.645886] ipq806x-gmac-dwmac 37000000.ethernet: IRQ eth_lpi not found
[    1.652178] ipq806x-gmac-dwmac 37000000.ethernet: PTP uses main clock
[    1.659109] ipq806x-gmac-dwmac 37000000.ethernet: User ID: 0x10, Synopsys ID: 0x37
[    1.664988] ipq806x-gmac-dwmac 37000000.ethernet: 	DWMAC1000
[    1.672360] ipq806x-gmac-dwmac 37000000.ethernet: DMA HW capability register supported
[    1.678283] ipq806x-gmac-dwmac 37000000.ethernet: RX Checksum Offload Engine supported
[    1.685985] ipq806x-gmac-dwmac 37000000.ethernet: COE Type 2
[    1.693797] ipq806x-gmac-dwmac 37000000.ethernet: TX Checksum insertion supported
[    1.699727] ipq806x-gmac-dwmac 37000000.ethernet: Wake-Up On Lan supported
[    1.707069] ipq806x-gmac-dwmac 37000000.ethernet: Enhanced/Alternate descriptors
[    1.713765] ipq806x-gmac-dwmac 37000000.ethernet: Enabled extended descriptors
[    1.721422] ipq806x-gmac-dwmac 37000000.ethernet: Ring mode enabled
[    1.728436] ipq806x-gmac-dwmac 37000000.ethernet: Enable RX Mitigation via HW Watchdog Timer
[    1.736736] ipq806x-gmac-dwmac 37200000.ethernet: IRQ eth_wake_irq not found
[    1.743201] ipq806x-gmac-dwmac 37200000.ethernet: IRQ eth_lpi not found
[    1.750469] ipq806x-gmac-dwmac 37200000.ethernet: PTP uses main clock
[    1.757241] ipq806x-gmac-dwmac 37200000.ethernet: User ID: 0x10, Synopsys ID: 0x37
[    1.763163] ipq806x-gmac-dwmac 37200000.ethernet: 	DWMAC1000
[    1.770708] ipq806x-gmac-dwmac 37200000.ethernet: DMA HW capability register supported
[    1.776555] ipq806x-gmac-dwmac 37200000.ethernet: RX Checksum Offload Engine supported
[    1.784166] ipq806x-gmac-dwmac 37200000.ethernet: COE Type 2
[    1.792119] ipq806x-gmac-dwmac 37200000.ethernet: TX Checksum insertion supported
[    1.797960] ipq806x-gmac-dwmac 37200000.ethernet: Wake-Up On Lan supported
[    1.805335] ipq806x-gmac-dwmac 37200000.ethernet: Enhanced/Alternate descriptors
[    1.812028] ipq806x-gmac-dwmac 37200000.ethernet: Enabled extended descriptors
[    1.819641] ipq806x-gmac-dwmac 37200000.ethernet: Ring mode enabled
[    1.826670] ipq806x-gmac-dwmac 37200000.ethernet: Enable RX Mitigation via HW Watchdog Timer
[    1.835003] i2c_dev: i2c /dev entries driver
[    1.845514] sdhci: Secure Digital Host Controller Interface driver
[    1.845709] sdhci: Copyright(c) Pierre Ossman
[    1.851693] sdhci-pltfm: SDHCI platform and OF driver helper
[    1.857382] krait-cache-scaling soc:l2-cache: error -ENODEV: _opp_set_regulators: no regulator (l2) found
[    1.863987] NET: Registered PF_INET6 protocol family
[    1.873094] Segment Routing with IPv6
[    1.876520] In-situ OAM (IOAM) with IPv6
[    1.880017] NET: Registered PF_PACKET protocol family
[    1.884278] 8021q: 802.1Q VLAN Support v1.8
[    1.889087] Registering SWP/SWPB emulation handler
[    1.939222] qcom_rpm 108000.rpm: RPM firmware 3.0.16777377
[    2.186091] qca8k 37000000.mdio-mii:10: configuring for fixed/rgmii link mode
[    2.196971] qca8k 37000000.mdio-mii:10: Link is Up - 1Gbps/Full - flow control off
[    2.203302] qca8k 37000000.mdio-mii:10: configuring for fixed/sgmii link mode
[    2.216861] qca8k 37000000.mdio-mii:10: Link is Up - 1Gbps/Full - flow control off
[    2.323085] qca8k 37000000.mdio-mii:10 lan1 (uninitialized): PHY [qca8k-0.0:00] driver [Qualcomm Atheros 8337 internal PHY] (irq=POLL)
[    2.453136] qca8k 37000000.mdio-mii:10 lan2 (uninitialized): PHY [qca8k-0.0:01] driver [Qualcomm Atheros 8337 internal PHY] (irq=POLL)
[    2.558510] qca8k 37000000.mdio-mii:10 lan3 (uninitialized): PHY [qca8k-0.0:02] driver [Qualcomm Atheros 8337 internal PHY] (irq=POLL)
[    2.673162] qca8k 37000000.mdio-mii:10 lan4 (uninitialized): PHY [qca8k-0.0:03] driver [Qualcomm Atheros 8337 internal PHY] (irq=POLL)
[    2.783032] qca8k 37000000.mdio-mii:10 wan (uninitialized): PHY [qca8k-0.0:04] driver [Qualcomm Atheros 8337 internal PHY] (irq=POLL)
[    2.784074] ipq806x-gmac-dwmac 37000000.ethernet eth0: entered promiscuous mode
[    2.794457] ipq806x-gmac-dwmac 37200000.ethernet eth1: entered promiscuous mode
[    2.801194] DSA: tree 0 setup
[    2.814550] thermal thermal_zone0: failed to read out thermal zone (-110)
[    2.825542] UBI: auto-attach mtd11
[    2.825575] ubi0: attaching mtd11
[    3.364422] ubi0: scanning is finished
[    3.375606] ubi0: attached mtd11 (name "ubi", size 64 MiB)
[    3.375641] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
[    3.379988] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
[    3.386930] ubi0: VID header offset: 2048 (aligned 2048), data offset: 4096
[    3.393704] ubi0: good PEBs: 512, bad PEBs: 0, corrupted PEBs: 0
[    3.400536] ubi0: user volume: 3, internal volumes: 1, max. volumes count: 128
[    3.406788] ubi0: max/mean erase counter: 2/0, WL threshold: 4096, image sequence number: 951216732
[    3.413759] ubi0: available PEBs: 0, total reserved PEBs: 512, PEBs reserved for bad PEB handling: 40
[    3.422812] ubi0: background thread "ubi_bgt0d" started, PID 69
[    3.423778] block ubiblock0_1: created from ubi0:1(?[    3.446180] List of all partitions:
[    3.446220] 1f00             256 mtdblock0 
[    3.446232]  (driver?)
[    3.452638] 1f01            1280 mtdblock1 
[    3.452647]  (driver?)
[    3.459149] 1f02            1280 mtdblock2 
[    3.459157]  (driver?)
[    3.465708] 1f03            2560 mtdblock3 
[    3.465719]  (driver?)
[    3.472168] 1f04            1152 mtdblock4 
[    3.472176]  (driver?)
[    3.478679] 1f05            1152 mtdblock5 
[    3.478688]  (driver?)
[    3.485271] 1f06            2560 mtdblock6 
[    3.485280]  (driver?)
[    3.491699] 1f07            2560 mtdblock7 
[    3.491708]  (driver?)
[    3.498260] 1f08            5120 mtdblock8 
[    3.498270]  (driver?)
[    3.504789] 1f09             512 mtdblock9 
[    3.504798]  (driver?)
[    3.511230] 1f0a            1280 mtdblock10 
[    3.511238]  (driver?)
[    3.518160] 1f0b           65536 mtdblock11 
[    3.518168]  (driver?)
[    3.524756] 1f0c             384 mtdblock12 
[    3.524765]  (driver?)
[    3.531283] 1f0d            1280 mtdblock13 
[    3.531292]  (driver?)
[    3.537935] 1f0e            2560 mtdblock14 
[    3.537945]  (driver?)
[    3.544477] 1f0f            1152 mtdblock15 
[    3.544557]  (driver?)
[    3.551075] 1f10            1152 mtdblock16 
[    3.551085]  (driver?)
[    3.557741] 1f11            2560 mtdblock17 
[    3.557750]  (driver?)
[    3.564268] 1f12            2560 mtdblock18 
[    3.564276]  (driver?)
[    3.570867] 1f13             384 mtdblock19 
[    3.570878]  (driver?)
[    3.577516] 1f14            5120 mtdblock20 
[    3.577525]  (driver?)
[    3.584060] 1f15           65536 mtdblock21 
[    3.584068]  (driver?)
[    3.590659] 1f16           94208 mtdblock22 
[    3.590668]  (driver?)
[    3.597333] 1f17           16384 mtdblock23 
[    3.597342]  (driver?)
[    3.603852] 1f18           16384 mtdblock24 
[    3.603862]  (driver?)
[    3.610450] fe00            3472 ubiblock0_1 
[    3.610459]  (driver?)
[    3.617125] No filesystem could mount root, tried: 
[    3.617132]  squashfs
[    3.619303] 
[    3.626560] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(31,23)
[    3.628083] CPU1: stopping
[    3.628091] CPU: 1 PID: 0 Comm: swapper/1 Not tainted 6.6.30 #0
[    3.628099] Hardware name: Generic DT based system
[    3.628110]  unwind_backtrace from show_stack+0x10/0x14
[    3.628143]  show_stack from dump_stack_lvl+0x40/0x4c
[    3.628165]  dump_stack_lvl from do_handle_IPI+0xec/0x120
[    3.628177]  do_handle_IPI from ipi_handler+0x18/0x20
[    3.628187]  ipi_handler from handle_percpu_devid_irq+0x78/0x134
[    3.628202]  handle_percpu_devid_irq from generic_handle_domain_irq+0x28/0x38
[    3.628223]  generic_handle_domain_irq from gic_handle_irq+0x74/0x88
[    3.628238]  gic_handle_irq from generic_handle_arch_irq+0x34/0x44
[    3.628254]  generic_handle_arch_irq from call_with_stack+0x18/0x20
[    3.628266]  call_with_stack from __irq_svc+0x80/0x98
[    3.628276] Exception stack(0xc1089f58 to 0xc1089fa0)
[    3.628282] 9f40:                                                       00000003 00000001
[    3.628288] 9f60: 000008bc 40000000 00000001 c0e04f68 c10e6000 c0e04fa4 00000000 00000000
[    3.628294] 9f80: 00000000 00000000 00000001 c1089fa8 c0a156b4 c0a15e1c 60000013 ffffffff
[    3.628299]  __irq_svc from default_idle_call+0x2c/0x30
[    3.628312]  default_idle_call from do_idle+0x1dc/0x22c
[    3.628327]  do_idle from cpu_startup_entry+0x28/0x2c
[    3.628338]  cpu_startup_entry from secondary_start_kernel+0x144/0x148
[    3.628350]  secondary_start_kernel from 0x423014f4
[    3.758712] Rebooting in 1 seconds..
```

Add the nor flash config to dts fix this bug.

```
[    3.136688] UBI: auto-attach mtd11
[    3.136721] ubi0: attaching mtd11
[    3.697193] ubi0: scanning is finished
[    3.708659] ubi0: attached mtd11 (name "ubi", size 64 MiB)
[    3.708692] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
[    3.713040] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
[    3.719977] ubi0: VID header offset: 2048 (aligned 2048), data offset: 4096
[    3.726808] ubi0: good PEBs: 512, bad PEBs: 0, corrupted PEBs: 0
[    3.733525] ubi0: user volume: 3, internal volumes: 1, max. volumes count: 128
[    3.739833] ubi0: max/mean erase counter: 2/0, WL threshold: 4096, image sequence number: 951216732
[    3.746862] ubi0: available PEBs: 0, total reserved PEBs: 512, PEBs reserved for bad PEB handling: 40
[    3.756080] ubi0: background thread "ubi_bgt0d" started, PID 78
[    3.766060] block ubiblock0_1: created from ubi0:1(rootfs)
[    3.770852] ubiblock: device ubiblock0_1 (rootfs) set to be root fi?[    3.788184] VFS: Mounted root (squashfs filesystem) readonly on device 254:0.
[    3.789080] Freeing unused kernel image (initmem) memory: 1024K
[    3.794525] Run /sbin/init as init process
```

### OpenWrt version

r26300-da0cd9d764

### OpenWrt release

SNAPSHOT

### OpenWrt target/subtarget

ipq806x/generic

### Device

Compex WPQ864

### Image kind

Official downloaded image

### Steps to reproduce

tftpboot openwrt-ipq806x-generic-compex_wpq864-initramfs-fit-uImage.itb
sysupgrade -n openwrt-ipq806x-generic-compex_wpq864-squashfs-nand-sysupgrade.bin

### Actual behaviour

_No response_

### Expected behaviour

_No response_

### Additional info

_No response_

### Diffconfig

_No response_

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

### PR DESCRIPTION
**Pull request by stargieg**
Add the nor flash config to dts.

Fixes: openwrt#15476

Signed-off-by: Patrick Grimm <patrick@lunatiki.de>

### CODE REVIEW

**Comment by john-tho** [comment]
If you are making related changes for a device, is it easier to review if all the commits are part of a single PR, especially if the changes interact.
If you could cherry-pick all your wpq864 commits to one of the branches for your open PRs, then force-push it and close the other two PRs, that should help, thanks.

File: target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-wpq864.dts [hunk 1]
```diff
@@ -404,8 +404,99 @@
...
+				};
+
+				partition@2100000 {
+					label = "1:APPSBL";
+					reg = <0x210000 0x80000>;
+					read-only;
+				};
+
+				partition@290000 {
+					label = "1:APPSBLENV";
```
Comment (john-tho) [inline target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-wpq864.dts:478]: You might run into problems with two `APPSBLENV`:

https://github.com/openwrt/openwrt/blob/4f87a4d84f3d6d1625962413ea56cca1ce83db7c/package/boot/uboot-envtools/files/ipq806x#L14-L15

**Comment by stargieg** [comment]
> If you are making related changes for a device, is it easier to review if all the commits are part of a single PR, especially if the changes interact. If you could cherry-pick all your wpq864 commits to one of the branches for your open PRs, then force-push it and close the other two PRs, that should help, thanks.

All PR are independent from each other.

File: target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-wpq864.dts (hunk 1 above)
Comment (stargieg) [reply target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-wpq864.dts:478]: I see. 
The first 3 parts of the fist matching line are process. That is 0:APPSBLENV
If the nand:0 flash is not available(corrupt,erased) then the copy 1:APPSBLENV
I from nand:1 can be used.
```
ubootenv_mtdinfo () {
	UBOOTENV_PART=$(cat /proc/mtd | grep APPSBLENV)
	mtd_dev=$(echo $UBOOTENV_PART | awk '{print $1}' | sed 's/:$//')
	mtd_size=$(echo $UBOOTENV_PART | awk '{print "0x"$2}')
	mtd_erase=$(echo $UBOOTENV_PART | awk '{print "0x"$3}')
```

**Comment by stargieg** [comment]
new PR

**Pull request closed without merging**